mixpanel/__init__.py

```diff
@@ -594,38 +596,36 @@ def send(self, endpoint, json_message, api_key=None, api_secret=None):
         self._write_request(self._endpoints[endpoint], json_message, api_key, api_secret)
 
     def _write_request(self, request_url, json_message, api_key=None, api_secret=None):
-        data = {
-            'data': json_message,
-            'verbose': 1,
-            'ip': 0,
-        }
-
         if isinstance(api_key, tuple):
             # For compatibility with subclassers, allow the auth details to be
             # packed into the existing api_key param.
             api_key, api_secret = api_key
 
+        params = {
+            'data': json_message,
+            'verbose': 1,
+            'ip': 0,
+        }
         if api_key:
-            data.update({'api_key': api_key})
-
-        headers = None
+            params['api_key'] = api_key
 
+        basic_auth = None
         if api_secret is not None:
-            headers = urllib3.util.make_headers(basic_auth="{}:".format(api_secret))
+            basic_auth = HTTPBasicAuth(api_secret, '')
 
         try:
-            response = self._http.request(
-                'POST',
+            response = self._session.post(
                 request_url,
-                fields=data,
-                headers=headers,
-                encode_multipart=False, # URL-encode payload in POST body.
+                json=params,
+                auth=basic_auth,
+                timeout=self._request_timeout,
+                verify=self._verify_cert,
             )
         except Exception as e:
             six.raise_from(MixpanelException(e), e)
 
         try:
-            response_dict = json.loads(response.data.decode('utf-8'))
+            response_dict = response.json()
```

Comment:
requests takes care of encoding issues?

Reply:
Yep, auto-sleuthed from the response headers. https://github.com/psf/requests/blob/1466ad713cf84738cd28f1224a7ab4a19e50e361/requests/models.py#L893

Reply:
wrong link: https://github.com/psf/requests/blob/1466ad713cf84738cd28f1224a7ab4a19e50e361/requests/models.py#L844-L854